lib/honeydew/worker.ex

```diff
@@ -175,8 +178,9 @@ defmodule Honeydew.Worker do
            Queue.worker_ready(queue_pid)
            :ok
 
-         {:error, e} ->
-           :ok = JobMonitor.job_failed(job_monitor, e)
+         {:error, %Crash{} = crash} ->
+           HoneydewLogger.job_failed(job, crash)
+           :ok = JobMonitor.job_failed(job_monitor, format_crash(crash))
```

Comment:
how about passing the `%Crash{}` through to the job monitor, instead of converting it to a tuple?

Reply:
That's a great idea. My goal was to not change the error format that the FailureHandlers were receiving, but I can move that logic into JobMonitor.

Reply:
Unfortunately the existing failure modes just pass along the reason, so anyone relying on the result of `{:error, reason} = Honeydew.yield(job)` would also be broken. This PR sets up a boundary setup to maintain the existing error API, which I think we should keep for the moment. I think if we want to rework how errors are returned, that could easily be part of a different commit.

I did just move the `format_crash_reason` logic into JobMonitor, which I like a lot better.